ios/app/brave_core_main.mm

```diff
@@ -52,11 +55,12 @@ @interface BraveCoreMain () {
 
 @implementation BraveCoreMain
 
-- (instancetype)init {
-  return [self initWithSyncServiceURL:@""];
+- (instancetype)initWithUserAgent:(NSString*)userAgent {
+  return [self initWithUserAgent:userAgent syncServiceURL:@""];
```

Comment:
Why do we remove this default constructor? By default if it builds debug it will use the debug URL, and if build release, then the prod URL.

Reply:
Nothing has changed with the sync service URL constructor, only change is that user agent is now required to be set at init so it is ready during WebMain initializer (which sets up default shared url loader configs)